### DISCUSSION TITLE
SIM800L library

### DISCUSSION THREAD

**Discussion by tadejales** [micropython/micropython#14256] [Libraries & Drivers]
Hello, 

I am self tought programer and I enjoy building things, but sometimes requirements are too large for my knowlage. Does anyone know how to add header to post request to be able to authenticate post request in this library. 

[https://github.com/pythings/Drivers](https://github.com/orgs/micropython/discussions/url)

I am succesfuly establishing GPRS communication from TTGO T-Call ESP32 board. I have also tested get and post requests and they work. But now I have problem establishing connection with MQTT to Adafruit IO server. I tried umqtt.simple but am getting OSError -202 which from what I was able to research means that MQTTClient is unable to establish connection. Credentials are correct as I have tested everything with cpython and I was able to post messages. Second options is to use direct posting over API but I have to authenticate myself so I need to add header to post request. 

I hope someone is able to help me out here.

**Comment by jotamudo** [comment]
If you take a look at some of the code the driver you sent provides on the [`execute_at_command`](https://github.com/pythings/Drivers/blob/d490b89ec9148d59692e8ff35a27c0feb57ce162/SIM800L.py#L135) function, there seems to be a command to fill in HTTP parameters (namely `AT+HTTPPARA={parameter},{value}`) so I'd guess you'd have to take a look at the AT commands of the modem to add your auth headers to the HTTP request.

The other thing you could try is using the [`dump_data`](https://github.com/pythings/Drivers/blob/d490b89ec9148d59692e8ff35a27c0feb57ce162/SIM800L.py#L139) one to just dump plain text data with the auth data and see if it forms it correctly

**Comment by chris35469** [comment]
Potential temporary fix for this error is mentioned in #13603
